Python/chapter01/1.8 - Zero Matrix/camilo_solution_1.8.py

```diff
@@ -2,21 +2,67 @@
 Zero Matrix: Write an algorithm such that if an element in an MxN matrix is 0,
 its entire row and column are set to 0.
 Hints:#17, #74, #702
+
+
+
+I've been stuck on this for a couple of days..
+Really good resource:
+https://www.youtube.com/watch?v=ZzpJgRvqSJQ&t=113s
+https://www.youtube.com/watch?v=qWeNXOCff3o&t=408s
 """
 
 
 
 def zeroMatrix(matrix):
-    m = len(matrix) - 1
-    n = len(matrix[0]) - 1
-    for i in range(m):
-        for j in range(n):
-                if matrix[i][j]  == 0:
-                    matrix[i+1][j] = 0
+    rowZero = False
```

Comment:
how would you solve it if instead of a matrix, you had to do it with a single array? i'm betting something like:

```python
hasZero = False
for value in array:
  if value == 0:
    hasZero = True
    break
# sidenote, the above can be expressed as
#   hasZero = any(value == 0 for value in array)
if hasZero:
  for i in len(array):
    array[i] = 0
```

since matrices are just a bunch of arrays stacked together, you should be able to repeat the above snippet for every row and column (with two separate `for` loops). so two loops with a snippet of code _very_ similar to the one above should suffice to solve this problem.